### PR TITLE
[backend] Add tags to override sqlc generated struct

### LINKED ISSUE #43: [Backend] Add overrides from column to tag for sqlc
As title.
為了最好的利用 sqlc 所產生的 struct, 必須為特定 column 新增額外的 tag 讓 echo.context.Bind() 能夠直接的將參數傳給 struct

### PR DESCRIPTION
**Pull request by x06lan**
As #43 
Also format some sql files add fix some syntax

### CODE REVIEW

**Comment by kryr34** [comment]
> ***pls only review sqlc.yaml***

then why do you need to merge other files

File: db/migrations/000001_db_setup.up.sql [hunk 1]
```diff
@@ -81,11 +85,13 @@ CREATE TABLE
         "image_id" UUID NOT NULL,
         PRIMARY KEY ("id", "version")
     );
+
 CREATE TABLE
     "coupon" (
         "id" SERIAL PRIMARY KEY,
         "type" coupon_type NOT NULL,
         "shop_id" INT NOT NULL,
+        "name" TEXT NOT NULL,
```
Comment (kryr34) [inline db/migrations/000001_db_setup.up.sql:94]: This file uses both `VARCHAR` and `TEXT`. Should it be made more consistent?
If further discussion is needed, I suggest deferring it to another PR.

**Review by Error0229 — DISMISSED** [review]
LGTM

File: .gitignore [hunk 2]
```diff
@@ -25,3 +25,4 @@ go.work
 
 *.env
 !.example.env
+twp
```
Comment (kryr34) [inline .gitignore]: No idea why is this needed

File: db/migrations/000001_db_setup.up.sql [hunk 3]
```diff
@@ -180,12 +187,15 @@ ADD
...
+    ) REFERENCES "product_archive" ("id", "version");
 
 ALTER TABLE "order_history"
 ADD
     FOREIGN KEY ("user_id") REFERENCES "user" ("id");
 
 ALTER TABLE "order_history"
 ADD
-    FOREIGN KEY ("shop_id") REFERENCES "shop" ("id");
+    FOREIGN KEY ("shop_id") REFERENCES "shop" ("id");
```
Comment (kryr34) [inline db/migrations/000001_db_setup.up.sql]: Is the trailing newline removed here by the formatter?

File: .gitignore (hunk 2 above)
Comment (x06lan) [reply .gitignore]: I need this

Comment (NOOBDY) [reply .gitignore]: I don't think this is needed

File: db/migrations/000001_db_setup.up.sql (hunk 1 above)
Comment (Error0229) [reply db/migrations/000001_db_setup.up.sql:94]: After some quick discussion, we use TEXT for any texts. Since it have higher flexibly and have no performance different/

**Review by NOOBDY — APPROVED** [review]
LGTM

**Review by Error0229 — APPROVED** [review]
🐱
